File: week02/ee/폴리오미노.py
import sys
sys.stdin = open("input.txt")

arr = list(input())

cnt = 0
board = [' ']*len(arr)

for i in range(len(arr)):
    if (arr[i] == 'X'):
        cnt += 1
    elif (arr[i] == '.'):
        board[i] = '.'

    if cnt == 4:
        for j in range(i-3, i+1):
            board[j] = 'A'
        cnt = 0
    elif arr[i] == '.' and cnt == 2:
        for j in range(i-2, i):
            board[j] = 'B'
        cnt = 0
    elif arr[i] == '.' and (cnt == 1 or cnt == 3):
        board = ['-1']
        cnt = 0
        break
    # print(board)
    # print(cnt)
else:
    i = i+1 # 마지막에서 인덱스값을 +1을 안 해줘서 계속 틀렸었다
    if cnt == 4:
        for j in range(i-3, i+1):
            board[j] = 'A'
    elif cnt == 2:
        for j in range(i-2, i):
            board[j] = 'B'
    elif cnt != 0:
        board = ['-1']

print(''.join(board))

File: week02/ee/폴리오미노.js
// 폴리오미노

const fs = require("fs");
const filePath = process.platform === "linux" ? "dev/stdin" : "./input.txt";

const input = fs.readFileSync(filePath).toString().trim();

let answer = input.replace(/XXXX/g, 'AAAA').replace(/XX/g, 'BB');

console.log(answer.split('').includes('X') ? -1 : answer)

// 이 문제를 파이썬으로 먼저 풀고 검색을 해봤었는데 
// 새까맣게 잊고 있었던 replace함수로 풀더라구요 ~~~

// 그래서 자스에도 replace가 있나 검색해봤는데 자스에도 있었고
// 문자열 내의 모든 단어를 치환하려면 정규식을 같이 써야한다고 합니다

// 정규식을 사용하는 방법: /패턴/플래그;
// 슬래시로 패턴을 감싼 뒤에 플래그를 끝에 추가해주는 형태
// 대표적인 플래그: i(대소문자 구별하지 않고 검색), g(문자열 내의 모든 패턴 검색)
// 패턴: 매칭하여 검색하고 싶은 문자열을 지정하는 것